src/browser/api_protocol/api_handlers/mesh_middleware.ts

```diff
@@ -128,6 +134,85 @@ function sendMessageMiddleware(msg: MessagePackage, next: () => void) {
     }
 }
 
+const handleExternalWindow = async (identity: Identity, toGroup: Identity): Promise<string|void> => {
```

Comment:
Shouldn't this be `: Promise<string|Error>`?

Reply:
moved error handling out of this fn, think it is much cleaner